src/ess/diffraction/filtering.py

```diff
@@ -67,3 +69,58 @@ def remove_bad_pulses(
     filtered = filtered.squeeze('good_pulse').copy(deep=False)
     del filtered.coords['good_pulse']
     return filtered
+
+
+def crop_tof(
+    data: RawData[RunType], tof_range: ValidTofRange
+) -> TofCroppedData[RunType]:
+    """Remove events outside the specified TOF range.
+
+    Parameters
+    ----------
+    data:
+        Data to be cropped.
+        Expected to have a coordinate called `'tof'`.
+    tof_range:
+        1d, len-2 variable containing the lower and upper bounds for
+        time-of-flight.
+
+    Returns
+    -------
+    :
+        Cropped data.
+    """
+    return TofCroppedData[RunType](
+        data.bin(tof=tof_range.to(unit=data.coords['tof'].unit))
+    )
```

Comment:
Note https://scipp.github.io/user-guide/binned-data/filtering.html#Extract-events-falling-into-a-parameter-interval, which would avoid the need for tof binning, i.e., adding `tof` as a dim.

Reply:
This doesn't work. It seems to do nothing:

![Screenshot_20231205_160814](https://github.com/scipp/essdiffraction/assets/11393224/bab79db8-bfdb-47ff-8760-15c21ea4b021)